### PR TITLE
cmd: Add options to set teams and and and remove PTO

### PR DESCRIPTION
**Pull request by twpayne**
This quick script makes updating `cilium-team-assignments.yaml` much quicker. No more editing YAML files, searching for Github usernames, or manually sorting lists.

Signed-off-by: Tom Payne <tom@isovalent.com>

### CODE REVIEW

**Comment by twpayne** [comment]
cc @aanm and @pchaigno as you both often edit this file.

File: cmd/team-updater/main.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	}
+
+	if dryRun {
+		return yaml.NewEncoder(os.Stdout).Encode(c)
+	}
+
+	return persistence.StoreState(configFilename, c)
+}
+
+func main() {
```
Comment (aanm) [inline cmd/team-updater/main.go:133]: Why not make this as an option for the team-management instead of a new binary?

Comment (twpayne) [reply cmd/team-updater/main.go]: OK, done. It makes the command a little more verbose to use.

File: cmd/main.go [hunk 2]
```diff
@@ -29,37 +30,41 @@ import (
...
-}
-
-var globalCtx, cancel = context.WithCancel(context.Background())
-
-func signals() {
-	signalCh := make(chan os.Signal, 1)
-	signal.Notify(signalCh, os.Interrupt)
-	<-signalCh
-	cancel()
 }
```
Comment (aanm) [inline cmd/main.go:64]: Why was this removed? :thinking:

**Review by aanm — CHANGES_REQUESTED** [review]

File: cmd/main.go (hunk 2 above)
Comment (twpayne) [reply cmd/main.go:64]: It breaks normal SIGINT handling. Normally Ctrl+C sends SIGINT which interrupts the process. With this signal handler Ctrl+C is ignored (it cancels the `globalCtx`, but this only has an effect if `globalCtx` has an operation in process). See the second commit in the PR for a bit more info.

Comment (aanm) [reply cmd/main.go:64]: Can't we fix the intended behavior instead? Aborting the program with Ctrl+C.

Comment (twpayne) [reply cmd/main.go:64]: The default Linux/Go behavior (i.e. without the bespoke `SIGINT` handling) is already the correct behavior. There is no need to trap `SIGINT`: the operating system will automatically clean up all associated resources.

Put differently: what are the reasons for including the bespoke `SIGINT` handling?

**Comment by twpayne** [comment]
@aanm Could you take another look please?

File: cmd/main.go (hunk 2 above)
Comment (aanm) [reply cmd/main.go:66]: @twpayne IIRC `SIGINT` wouldn't interrupt the program without this, i.e., the github requests would still keep going.

**Review by aanm — APPROVED** [review]

File: cmd/main.go (hunk 2 above)
Comment (twpayne) [reply cmd/main.go]: OK, I've removed this commit.
